sky/serve/controller.py

```diff
@@ -89,6 +92,37 @@ def _run_autoscaler(self):
                     logger.error(f'  Traceback: {traceback.format_exc()}')
             time.sleep(self._autoscaler.get_decision_interval())
 
+    def _purge_replica(self, replica_id: int) -> Dict[str, str]:
+        logger.info(f'Purging replica {replica_id}...')
+        replica_info = serve_state.get_replica_info_from_id(
+            self._service_name, replica_id)
+        assert replica_info is not None
+        replica_cluster_is_remaining = False
+        if replica_info.status in serve_state.ReplicaStatus.failed_statuses():
+            if global_user_state.get_cluster_from_name(
+                    replica_info.cluster_name) is not None:
+                replica_cluster_is_remaining = True
+            serve_state.remove_replica(self._service_name, replica_id)
+            if replica_cluster_is_remaining:
+                return {
+                    'message':
+                        f'{colorama.Fore.YELLOW}Purged replica {replica_id} '
+                        f'with failed status ({replica_info.status}). This may'
+                        f' indicate a resource leak. Please check the following'
+                        f' SkyPilot cluster on the controller: '
+                        f'{replica_info.cluster_name}{colorama.Style.RESET_ALL}'
+                }
```

Comment:
as said in the comment above, we should cleanup this cluster as well. how about adding an argument `purge: bool` in `ReplicaManager.scale_down` and remove the replica record here?
https://github.com/skypilot-org/skypilot/blob/e51c2fb4b7e5f06cde6070ad0c758df3bc888026/sky/serve/replica_managers.py#L752-L775